enterprise/frontend/src/embedding-sdk/types.ts

```diff
@@ -1,14 +1,25 @@
-type SDKAuthType =
-  | {
-      authType: "apiKey";
-      apiKey: string;
-    }
-  | {
-      authType: "jwt";
-      jwtProviderUri: string;
-    };
-
-export type SDKConfigType = {
+type JWTAuthType = {
+  authType: "jwt";
+  jwtProviderUri: string;
+};
+
+type ApiKeyAuthType = {
+  authType: "apiKey";
+  apiKey: string;
+};
+
+type BaseSDKConfigType = {
```

Comment:
I've been trying to get the types to work on `use-init-data` and its tests for the error states, since there is the distinct possibility that someone might put a bad config. I don't want to cast everything so I figured using this `BaseSdkConfigType` would be helpful, but I'm not totally sure if this is the right way to go about this issue

Reply:
I think it's fine leaving this as is for now just to not block this PR from shipping. But I'd still prefer the type to be more strict like before. I'm not sure how to make that work with the current set up we have.

Maybe we need another layer that validates the config and in that layer the config type could be more relaxed, if it passes that test then we could assume the `config` type in `use-init-data` would be the correct JWT or API key config type. I'm still not sure how that would work in practice though. But it feels like we're mixing type validation and runtime validation in the same file.

Reply:
Yep, I totally agree. I'll push this PR for now but I'll think about it a little more. I like the idea of only allowing valid configs for any component that's below the MetabaseProvider (or wherever we decide to validate the config)